js/derivadex.js

```diff
@@ -0,0 +1,599 @@
+        //             "value": {
+        //                 "tickSize": "0.1",
+        //                 "minOrderSize": "0.0001",
+        //                 "maxOrderNotional": "0",
+        //                 "maxTakerPriceDeviation": "0.02"
+        //             }
+        //         },
+        //         {
+        //             "kind": 0,
+        //             "name": "BTCPERP",
+        //             "expr": "\n(Market :name "BTCPERP"\n :tick-size 1\n :max-order-notional 1000000\n :max-taker-price-deviation 0.02\n :min-order-size 0.00001\n)",
+        //             "value": {
+        //                 "tickSize": "1",
+        //                 "minOrderSize": "0.00001",
+        //                 "maxOrderNotional": "1000000",
+        //                 "maxTakerPriceDeviation": "0.02"
+        //             }
+        //         },
+        //         {
+        //             "kind": 0,
+        //             "name": "ETHPERP",
+        //             "expr": "\n(Market :name "ETHPERP"\n :tick-size 0.1\n :max-order-notional 1000000\n :max-taker-price-deviation 0.02\n :min-order-size 0.0001\n)",
+        //             "value": {
+        //                 "tickSize": "0.1",
+        //                 "minOrderSize": "0.0001",
+        //                 "maxOrderNotional": "1000000",
+        //                 "maxTakerPriceDeviation": "0.02"
+        //             }
+        //         }
+        //     ],
+        //         "timestamp": 1674260369,
+        //         "success": true
+        // }
+        let markets = response['value'];
+        markets = markets.filter ((market) => market['name'] !== 'DDXPERP');
+        return markets.map ((market) => {
+            const name = market['name'];
+            const base = name.slice (0, -4);
+            return {
+                'id': name,
+                'symbol': name,
+                'base': base,
+                'quote': 'USD',
+                'settle': 'USDC',
+                'baseId': base.toLowerCase (),
+                'quoteId': 'usd',
+                'settleId': 'usdc',
+                'type': 'swap',
+                'spot': 'false',
+                'margin': 'false',
+                'swap': 'true',
+                'future': 'false',
+                'option': 'swap',
+                'active': 'true',
+                'contract': 'true',
+                'linear': 'true',
+                'inverse': 'false',
+                'taker': '0.002',
+                'maker': '0.000',
+                'precision': {
+                    'amount': 6,
+                    'price': 6,
+                    'quote': 6,
+                },
+                'limits': {
+                    'leverage': {
+                        'min': 1,
+                        'max': 3,
+                    },
+                    'amount': {
+                        'min': market['value']['tickSize'],
```

Comment:
even if these fields are for the precision of amount and price, I dont think tickSize would below in the amount category. For example tickSize is 1 for the BTC market. That doesnt make much sense in the context of an amount (whereas the min order size of .00001 does)